File: FitBook.Model/Requests/Halls/HallInsertRequest.cs
namespace FitBook.Model.Requests.Halls;

public class HallInsertRequest
{
    public string Name { get; set; } = string.Empty;
    public int Capacity { get; set; }
    public string? LocationDescription { get; set; }
    public bool IsActive { get; set; }
}

File: UI/fitbook_desktop/lib/models/requests/hall_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

import '../common/api_request_body.dart';

part 'hall_insert_request.g.dart';

/// Mirrors `FitBook.Model.Requests.HallInsertRequest`.
@JsonSerializable()
class HallInsertRequest implements ApiRequestBody {
  HallInsertRequest({
    required this.name,
    required this.capacity,
    this.locationDescription,
    required this.isActive,
  });

  final String name;
  final int capacity;
  final String? locationDescription;
  final bool isActive;

  factory HallInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$HallInsertRequestFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$HallInsertRequestToJson(this);
}
